### PR TITLE
Update: change reporting location for `curly` rule (refs #12334)

### PR DESCRIPTION
**Pull request by snitin315**
#### Prerequisites checklist

- [X] I have read the [contributing guidelines](https://github.com/eslint/eslint/blob/master/CONTRIBUTING.md).

#### What is the purpose of this pull request? (put an "X" next to an item)

[ ] Documentation update
[ ] Bug fix ([template](https://raw.githubusercontent.com/eslint/eslint/master/templates/bug-report.md))
[ ] New rule ([template](https://raw.githubusercontent.com/eslint/eslint/master/templates/rule-proposal.md))
[X] Changes an existing rule ([template](https://raw.githubusercontent.com/eslint/eslint/master/templates/rule-change-proposal.md))
[ ] Add autofixing to a rule
[ ] Add a CLI option
[ ] Add something to the core
[ ] Other, please explain:

#### What changes did you make? (Give an overview)

Refers https://github.com/eslint/eslint/issues/12334

Fixes https://github.com/eslint/eslint/pull/13282

#### Is there anything you'd like reviewers to focus on?
No

### CODE REVIEW

**Comment by mdjermanovic** [comment]
Now I think that highlighting the body/consequent/alternate node would make the most sense, as that's the part of the code that should be fixed. The rule reports "Expected '{' after 'while' condition.", but the whole body should be wrapped in a block. A similar rule `arrow-body-style` reports `body.loc` as the error location.

**Comment by snitin315** [comment]
Makes sense, I will update 👍

**Comment by snitin315** [comment]
@mdjermanovic updated 👍🏻

File: lib/rules/curly.js [hunk 1]
```diff
@@ -359,19 +421,23 @@ module.exports = {
                 check() {
                     if (this.expected !== null && this.expected !== this.actual) {
                         if (this.expected) {
+                            const loc = getReportingLoc(node, name);
+
                             context.report({
                                 node,
-                                loc: (name !== "else" ? node : getElseKeyword(node)).loc.start,
+                                loc,
```
Comment (mdjermanovic) [inline lib/rules/curly.js:428]:
```suggestion
                                loc: body.loc,
```

I think there's no need for `function getReportingLoc`, we can just use the `body` param instead. That's the body/consequent/alternate node being currently checked.

Comment (snitin315) [reply lib/rules/curly.js]: Yeah, now we can directly use `body.loc`, updated 👍🏻

**Review by mdjermanovic — APPROVED** [review]
LGTM, thanks!

**Pull request merged**